e2e/pageobjects/ide/Editor.ts

```diff
@@ -13,44 +13,74 @@ import { DriverHelper } from '../../utils/DriverHelper';
 import { CLASSES } from '../../inversify.types';
 import { TestConstants } from '../../TestConstants';
 import { By, Key } from 'selenium-webdriver';
+import { error } from 'selenium-webdriver';
 
 @injectable()
 export class Editor {
+    private static readonly SUGGESTION_WIDGET_BODY_CSS: string = 'div.visible[widgetId=\'editor.widget.suggestWidget\']';
 
-    private static readonly EDITOR_BODY_CSS: string = '#theia-main-content-panel .lines-content';
-    private static readonly SUGGESTION_WIDGET_BODY_CSS: string = 'div[widgetId=\'editor.widget.suggestWidget\']';
-    private static readonly EDITOR_INTERACTION_СSS: string = '.monaco-editor textarea';
     constructor(@inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) { }
 
-    async waitSuggestionContainer(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+    public async waitSuggestionContainer(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
         await this.driverHelper.waitVisibility(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS), timeout);
     }
 
-    async waitSuggestionContainerClosed(attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS, polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
-        await this.driverHelper.waitDisappearance(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS), attempts, polling);
+    public async waitSuggestionContainerClosed(attempts: number = TestConstants.TS_SELENIUM_DEFAULT_ATTEMPTS,
+        polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING) {
+        await this.driverHelper.waitDisappearanceTestWithTimeout(By.css(Editor.SUGGESTION_WIDGET_BODY_CSS));
     }
 
-    async waitSuggestion(suggestionText: string, timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
-        await this.driverHelper.waitVisibility(By.xpath(this.getSuggestionLineXpathLocator(suggestionText)), timeout);
+    public async waitSuggestion(editorTabTitle: string,
+        suggestionText: string,
+        timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
+
+        const suggestionLocator: By = By.xpath(this.getSuggestionLineXpathLocator(suggestionText));
+
+        await this.driverHelper.getDriver().wait(async () => {
+            await this.waitSuggestionContainer();
+            try {
+                await this.driverHelper.waitVisibility(suggestionLocator, 5000);
+                return true;
+            } catch (err) {
+                const isTimeoutError: boolean = err instanceof error.TimeoutError;
+                if (!isTimeoutError) {
+                    throw err;
+                }
+
+                await this.pressEscapeButton(editorTabTitle);
```

Comment:
Why not to call ` performKeyCombination ` directly from here?

Reply:
Answer 2 https://github.com/eclipse/che/pull/13479#discussion_r290705855